File: 05/Code/Java/SymbolTable.java
/*
 *A symbol table that keeps a correspondence between
        symbolic labels and numeric addresses
 */
class SymbolTable
{

    /*
     * Creates the initial symbol table
     */
    public SymbolTable()
    {
    }
    
    /*
     *Adds the pair (symbol, address) to the table.
     */ 
    public void addEntry(String symbol, int address)
    {
    }
    
    /*
     * Adds the symbol to the table, setting the address
     *  to the next availble memory location 
     */
    public void addEntry(String symbol)
    {
    }

    /*
     * Does the symbol table contain the given symbol?
     */
    public boolean contains(String symbol)
    {
        return false;
    }

    /*
     * Returns the integer address associated with the symbol
     */
    public int getAddress(String symbol)
    {
        return -1;
    }
}

File: 05/Code/Python/symbol_table.py
class SymbolTable:
    """A symbol table that keeps a correspondence between
        symbolic labels and numeric addresses"""

    def __init__(self):
        """Creates the initial symbol table"""
        None
        
    def add_entry(self, symbol, address=None):
        """Adds the pair (symbol, address) to the table.
            If address is None, sets address to the next
            available memory location"""
        None

    def contains(self, symbol):
        """Does the symbol table contain the given symbol?
            Returns boolean"""
        None

    def get_address(self, symbol):
        """Returns the integer address associated with the symbol"""
        None

